In [1]:
import pandas as pd
import numpy as np
import requests
import json

# CLIMATOLOGÍA

La comprensión de la climatología de las distintas áreas será un aspecto clave al momento de establecer la empresa, dado que el tipo de producto que ofrece nuestro cliente no puede ubicarse en cualquier región debido a los siguientes factores:

Tipo de techo: Los techos son de cubierta plana, lo que implica que no son aptos para soportar climas extremadamente fríos, con nieve, o lluvias intensas, ya que estas condiciones pueden comprometer su funcionalidad.

Aislamiento térmico: Aunque las viviendas cuentan con un sistema de aislamiento térmico, este está diseñado para resistir temperaturas moderadas, por lo que no es adecuado para zonas con variaciones térmicas extremas.

Con base en estas consideraciones, limitaremos nuestra búsqueda a aquellas regiones donde el clima no resulte un factor de riesgo para la vivienda. Las zonas que analizaremos serán las anteriormente nombras en el df 1_precio_metro_cuadrado: Cantabria, Galicia, Navarra y Andalucía.

Para ello, utilizaremos los datos proporcionados por la AEMET de estas áreas para realizar un análisis detallado.

## 1. CANTABRIA

Para la obtención de los datos se ha accedido a la API de AEMET, para ello, se ha solicitado una API KEY, una vez obtenida, hemos filtrado los datos por Comunidad Autonoma.

Se ha tenido en cuenta la obtención de los datos del 2023 para poder tener los 12 meses del año

### 1.1. Obtención de los datos: solicitud GET a la API de AEMET

In [2]:
 #Realizamos una solicitud GET a la API de AEMET para obtener datos climatológicos
cantabria1 = requests.get ("https://opendata.aemet.es/opendata/api/valores/climatologicos/mensualesanuales/datos/anioini/2023/aniofin/2023/estacion/1111X/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwYXVsYWFqczIwQGdtYWlsLmNvbSIsImp0aSI6ImQ3OTIwMjgzLTUyMTItNDcwNy1hM2ViLTI5NGUzMzMwODFkMiIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNzMwNDc5NjAxLCJ1c2VySWQiOiJkNzkyMDI4My01MjEyLTQ3MDctYTNlYi0yOTRlMzMzMDgxZDIiLCJyb2xlIjoiIn0.vYC1-je9UixqSOoeH638C_vyHqNsozzCBd8HBsXG7VE")

#Convertimos la respuesta de la solicitud en formato JSON
cantabria1.json()

#Realizamos una segunda solicitud GET utilizando la URL obtenida de la primera respuesta, ya que la primera url expira a los pocos minutos, por lo que acemos la solicitud a "datos"
cantabria_1 = requests.get(cantabria1.json().get("datos"))

#Convertimos la respuesta de la solicitud en formato JSON
cantabria = cantabria_1.json()
cantabria

[{'fecha': '2023-10',
  'indicativo': '1111X',
  'p_max': '21.2(13)',
  'n_cub': '12',
  'hr': '77',
  'n_gra': '0',
  'n_fog': '3',
  'inso': '5.6',
  'q_max': '1023.3(03)',
  'nw_55': '10',
  'q_mar': '1012.0',
  'q_med': '1005.1',
  'tm_min': '16.2',
  'ta_max': '31.8(02)',
  'ts_min': '22.5',
  'nt_30': '1',
  'nv_0050': '0',
  'n_des': '4',
  'w_racha': '26/27.8(20)',
  'np_100': '5',
  'n_nub': '15',
  'p_sol': '51',
  'nw_91': '1',
  'ts_20': '18.7',
  'np_001': '16',
  'ta_min': '11.7(24)',
  'w_rec': '399',
  'e': '167',
  'np_300': '0',
  'nv_1000': '1',
  'evap': '1399',
  'p_mes': '131.5',
  'n_llu': '19',
  'n_tor': '0',
  'w_med': '17',
  'nt_00': '0',
  'ti_max': '17.3',
  'n_nie': '0',
  'tm_mes': '19.1',
  'tm_max': '22.1',
  'nv_0100': '1',
  'q_min': '974.9(20)',
  'np_010': '14'},
 {'fecha': '2023-11',
  'indicativo': '1111X',
  'p_max': '29.4(27)',
  'n_cub': '15',
  'hr': '81',
  'n_gra': '0',
  'n_fog': '2',
  'inso': '3.2',
  'q_max': '1026.5(23)',
  'nw_55': '1

### 1.2 Filtrado de los datos mediante un archivo JSON

Una vez nos devuelve un archivo json, seleccionamos los datos necesarios. Creamos una lista vacia donde iremos añadiedno dichos datos.

¡A tener en cuenta! Viendo los datos, nos hemos dado cuenta de en las fechas el valor 2023 - 13 es el total de cada mez, por lo que hemos descartado dicho mes para nuestro análisis

In [3]:
fechas = []
temp_max_cantabria = []
temp_min_cantabria = []
precipitacion_max_cantabria = []


for elemento in cantabria:
    if elemento["fecha"] != "2023-13":
        fechas.append(elemento["fecha"])
        temp_max_cantabria.append(elemento["tm_max"])
        temp_min_cantabria.append(elemento["tm_min"])
        precipitacion_max_cantabria.append(elemento["p_max"])


### 1.3 Limpieza de los datos

Tras obtener los datos, limpiamos "precipitaciones" y nos quedamos con lo que está antes del paréntesis.

In [4]:
precipitacion_max_cantabria = [elemento.split('(')[0] for elemento in precipitacion_max_cantabria]

## 2. GALICIA

Hacemos las mismas modificaciones para las siguientes comunidades

### 2.1 Obtención de los datos: solicitud GET a la API de AEMET

In [5]:
galicia = requests.get("https://opendata.aemet.es/opendata/api/valores/climatologicos/mensualesanuales/datos/anioini/2023/aniofin/2023/estacion/1387/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwYXVsYWFqczIwQGdtYWlsLmNvbSIsImp0aSI6ImQ3OTIwMjgzLTUyMTItNDcwNy1hM2ViLTI5NGUzMzMwODFkMiIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNzMwNDc5NjAxLCJ1c2VySWQiOiJkNzkyMDI4My01MjEyLTQ3MDctYTNlYi0yOTRlMzMzMDgxZDIiLCJyb2xlIjoiIn0.vYC1-je9UixqSOoeH638C_vyHqNsozzCBd8HBsXG7VE")
galicia.json()

galicia_1 =requests.get(galicia.json().get("datos"))
galicia_datos = galicia_1.json()
galicia_datos

[{'fecha': '2023-10',
  'indicativo': '1387',
  'p_max': '32.6(19)',
  'n_cub': '11',
  'glo': '28536',
  'hr': '75',
  'n_gra': '1',
  'n_fog': '5',
  'inso': '4.7',
  'q_max': '1021.4(03)',
  'nw_55': '13',
  'q_mar': '1011.3',
  'q_med': '1003.5',
  'tm_min': '15.6',
  'ta_max': '33.4(07)',
  'ts_min': '20.0',
  'nt_30': '1',
  'nv_0050': '0',
  'n_des': '5',
  'w_racha': '30/27.5(20)',
  'np_100': '9',
  'n_nub': '15',
  'p_sol': '42',
  'nw_91': '2',
  'ts_20': '18.9',
  'np_001': '20',
  'ta_min': '10.9(30)',
  'w_rec': '322',
  'e': '162',
  'np_300': '1',
  'nv_1000': '4',
  'evap': '915',
  'p_mes': '219.6',
  'n_llu': '19',
  'n_tor': '3',
  'ts_10': '18.7',
  'w_med': '13',
  'nt_00': '0',
  'ti_max': '16.6',
  'n_nie': '0',
  'tm_mes': '18.8',
  'tm_max': '22.0',
  'nv_0100': '0',
  'ts_50': '19.5',
  'q_min': '977.3(19)',
  'np_010': '19'},
 {'fecha': '2023-11',
  'indicativo': '1387',
  'p_max': '36.4(05)',
  'n_cub': '12',
  'glo': '16599',
  'hr': '80',
  'n_gra': '0',


### 2.2 Filtrado de los datos mediante un archivo JSON

In [6]:
temp_max_galicia = []
temp_min_galicia = []
precipitacion_max_galicia = []

for elemento in galicia_datos:
    if elemento["fecha"] != "2023-13":
        temp_max_galicia.append(elemento["tm_max"])
        temp_min_galicia.append(elemento["tm_min"])
        precipitacion_max_galicia.append(elemento["p_max"])

### 2.3 Limpieza de los datos

In [7]:
precipitacion_max_galicia = [elemento.split('(')[0] for elemento in precipitacion_max_galicia]


## 3. NAVARRA

### 3.1 Obtención de los datos: solicitud GET a la API de AEMET

In [8]:
navarra1 = requests.get("https://opendata.aemet.es/opendata/api/valores/climatologicos/mensualesanuales/datos/anioini/2023/aniofin/2023/estacion/9263D/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwYXVsYWFqczIwQGdtYWlsLmNvbSIsImp0aSI6ImQ3OTIwMjgzLTUyMTItNDcwNy1hM2ViLTI5NGUzMzMwODFkMiIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNzMwNDc5NjAxLCJ1c2VySWQiOiJkNzkyMDI4My01MjEyLTQ3MDctYTNlYi0yOTRlMzMzMDgxZDIiLCJyb2xlIjoiIn0.vYC1-je9UixqSOoeH638C_vyHqNsozzCBd8HBsXG7VE")
navarra1.json()

navarra_1 = requests.get(navarra1.json().get("datos"))
navarra = navarra_1.json()
navarra

[{'fecha': '2023-10',
  'indicativo': '9263D',
  'p_max': '21.2(29)',
  'n_cub': '10',
  'glo': '33726',
  'hr': '69',
  'n_gra': '0',
  'n_fog': '0',
  'inso': '6.1',
  'q_max': '975.7(03)',
  'nw_55': '5',
  'q_mar': '1013.2',
  'q_med': '961.2',
  'tm_min': '10.8',
  'ta_max': '32.5(01)',
  'ts_min': '17.6',
  'nt_30': '5',
  'nv_0050': '0',
  'n_des': '8',
  'w_racha': '17/20.6(17)',
  'np_100': '4',
  'n_nub': '13',
  'p_sol': '54',
  'nw_91': '0',
  'np_001': '12',
  'ta_min': '4.7(22)',
  'w_rec': '240',
  'e': '136',
  'np_300': '0',
  'nv_1000': '0',
  'p_mes': '86.0',
  'n_llu': '13',
  'n_tor': '0',
  'w_med': '10',
  'nt_00': '0',
  'ti_max': '16.0',
  'n_nie': '0',
  'tm_mes': '17.2',
  'tm_max': '23.6',
  'nv_0100': '0',
  'q_min': '933.4(19)',
  'np_010': '11'},
 {'fecha': '2023-11',
  'indicativo': '9263D',
  'p_max': '29.7(30)',
  'n_cub': '11',
  'glo': '17357',
  'hr': '79',
  'n_gra': '0',
  'n_fog': '2',
  'inso': '3.8',
  'q_max': '976.9(23)',
  'nw_55': '5',
  'q

### 3.2 Filtrado de los datos mediante un archivo JSON

In [9]:
temp_max_navarra = []
temp_min_navarra = []
precipitacion_max_navarra= []

for elemento in navarra:
    if elemento["fecha"] != "2023-13":
        temp_max_navarra.append(elemento["tm_max"])
        temp_min_navarra.append(elemento["tm_min"])
        precipitacion_max_navarra.append(elemento["p_max"])

### 3.3 Limpieza de los datos

In [10]:
precipitacion_max_navarra = [elemento.split('(')[0] for elemento in precipitacion_max_navarra]

## 4. ANDALUCIA

### 4.1 Obtención de los datos: solicitud GET a la API de AEMET

In [11]:
sevilla1 =requests.get("https://opendata.aemet.es/opendata/api/valores/climatologicos/mensualesanuales/datos/anioini/2023/aniofin/2023/estacion/4642E/?api_key=eyJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJwYXVsYWFqczIwQGdtYWlsLmNvbSIsImp0aSI6ImQ3OTIwMjgzLTUyMTItNDcwNy1hM2ViLTI5NGUzMzMwODFkMiIsImlzcyI6IkFFTUVUIiwiaWF0IjoxNzMwNDc5NjAxLCJ1c2VySWQiOiJkNzkyMDI4My01MjEyLTQ3MDctYTNlYi0yOTRlMzMzMDgxZDIiLCJyb2xlIjoiIn0.vYC1-je9UixqSOoeH638C_vyHqNsozzCBd8HBsXG7VE")
sevilla1.json()

sevilla_1 = requests.get(sevilla1.json().get("datos"))
andalucia = sevilla_1.json()
andalucia

[{'fecha': '2023-10',
  'indicativo': '4642E',
  'p_max': '67.6(22)',
  'n_cub': '9',
  'glo': '42379',
  'hr': '71',
  'n_gra': '3',
  'n_fog': '3',
  'inso': '7.5',
  'q_max': '1022.0(04)',
  'nw_55': '3',
  'q_mar': '1015.2',
  'q_med': '1012.9',
  'tm_min': '16.2',
  'ta_max': '34.6(05)',
  'ts_min': '20.6',
  'nt_30': '10',
  'nv_0050': '0',
  'n_des': '8',
  'w_racha': '01/28.9(22)',
  'np_100': '3',
  'n_nub': '14',
  'p_sol': '66',
  'nw_91': '1',
  'np_001': '10',
  'ta_min': '10.2(24)',
  'w_rec': '261',
  'e': '179',
  'np_300': '3',
  'nv_1000': '2',
  'evap': '1013',
  'p_mes': '173.6',
  'n_llu': '13',
  'n_tor': '0',
  'w_med': '12',
  'nt_00': '0',
  'ti_max': '16.4',
  'n_nie': '0',
  'tm_mes': '21.5',
  'tm_max': '26.8',
  'nv_0100': '0',
  'q_min': '984.5(22)',
  'np_010': '9'},
 {'fecha': '2023-11',
  'indicativo': '4642E',
  'p_max': '14.6(30)',
  'n_cub': '4',
  'glo': '32801',
  'hr': '75',
  'n_gra': '0',
  'n_fog': '1',
  'inso': '7.4',
  'q_max': '1025.4(07)',

### 4.2 Filtrado de los datos mediante un archivo JSON

In [12]:
temp_max_andalucia = []
temp_min_andalucia = []
precipitacion_max_andalucia = []


for elemento in andalucia:
    if elemento["fecha"] != "2023-13":
        temp_max_andalucia.append(elemento["tm_max"])
        temp_min_andalucia.append(elemento["tm_min"])
        precipitacion_max_andalucia.append(elemento["p_max"])

### 4.3 Limpieza de los datos

In [13]:
precipitacion_max_andalucia = [elemento.split('(')[0] for elemento in precipitacion_max_andalucia]

## 5. CREACIÓN DEL DF CON LOS DATOS OBTENIDOS

### 5.1 Creamos el DF

In [14]:
#Cremos el DataFrame con las listas de cada comunidad.

df_1 = pd.DataFrame({ "fechas" :fechas,
                   "temp_max_cantabria": temp_max_cantabria,
                   "temp_min_cantabria" : temp_min_cantabria,
                   "precipitaciones_cantabria":precipitacion_max_cantabria,
                   "temp_max_galicia": temp_max_galicia,
                   "temp_min_galicia":temp_min_galicia,
                   "precipitaciones_galicia": precipitacion_max_galicia,
                   "temp_max_navarra":temp_max_navarra,
                   "temp_min_navarra":temp_min_navarra,
                   "precipitaciones_navarra":precipitacion_max_navarra,
                   "temp_max_andalucia": temp_max_andalucia,
                   "temp_min_andalucia": temp_min_andalucia,
                   "precipitaciones_andalucia": precipitacion_max_andalucia
})

In [15]:
df_1

fechas temp_max_cantabria temp_min_cantabria precipitaciones_cantabria  \
0   2023-10               22.1               16.2                      21.2   
1   2023-11               17.5               12.1                      29.4   
2   2023-12               14.5                9.7                      16.6   
3    2023-1               13.1                7.7                      94.6   
4    2023-2               12.7                7.6                      45.2   
5    2023-3               16.9               10.4                      17.0   
6    2023-4               15.8               11.0                      10.5   
7    2023-5               18.1               13.5                      14.0   
8    2023-6               22.0               17.5                       9.7   
9    2023-7               23.0               18.1                       8.2   
10   2023-8               23.7               18.5                      17.8   
11   2023-9               24.0               18.3                      60.2   

   temp_max_galicia temp_min_galicia precipitaciones_galicia temp_max_navarra  \
0              22.0             15.6                    32.6             23.6   
1              17.2             12.3                    36.4             15.3   
2              14.8              9.8                    56.8             11.0   
3              13.7              8.9                    22.0              8.4   
4              15.0              7.1                    11.2             11.8   
5              16.9             11.0                    11.4             17.7   
6              18.8             11.7                    12.2             19.8   
7              20.3             14.1                     3.8             21.5   
8              23.2             16.9                    17.8             27.1   
9              23.6             17.2                     4.2             29.7   
10             24.7             18.3                     3.8             31.8   
11             24.4             17.4                    16.8             27.4   

   temp_min_navarra precipitaciones_navarra temp_max_andalucia  \
0              10.8                    21.2               26.8   
1               6.8                    29.7               22.2   
2               2.4                     9.6               17.6   
3               0.6                    30.6               17.2   
4              -1.5                    21.9               18.4   
5               3.7                     4.1               22.7   
6               5.9                    12.4               27.9   
7               8.6                    34.4               27.3   
8              14.3                    12.3               31.0   
9              15.1                    29.7               34.3   
10             15.8                     8.4               36.6   
11             13.9                    62.6               29.0   

   temp_min_andalucia precipitaciones_andalucia  
0                16.2                      67.6  
1                11.2                      14.6  
2                 7.0                       5.2  
3                 6.4                       5.0  
4                 6.3                      29.8  
5                 9.4                       4.2  
6                12.3                       1.0  
7                15.1                      11.4  
8                18.9                      18.6  
9                20.5                       0.0  
10               21.0                       0.0  
11               17.3                       6.2

### 5.2 Limpieza del DF

Observamos como nos devuelve el dataframe, pero los valores los catalagola como object, lo pasamos a float

In [16]:
df_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 13 columns):
 #   Column                     Non-Null Count  Dtype 
---  ------                     --------------  ----- 
 0   fechas                     12 non-null     object
 1   temp_max_cantabria         12 non-null     object
 2   temp_min_cantabria         12 non-null     object
 3   precipitaciones_cantabria  12 non-null     object
 4   temp_max_galicia           12 non-null     object
 5   temp_min_galicia           12 non-null     object
 6   precipitaciones_galicia    12 non-null     object
 7   temp_max_navarra           12 non-null     object
 8   temp_min_navarra           12 non-null     object
 9   precipitaciones_navarra    12 non-null     object
 10  temp_max_andalucia         12 non-null     object
 11  temp_min_andalucia         12 non-null     object
 12  precipitaciones_andalucia  12 non-null     object
dtypes: object(13)
memory usage: 1.3+ KB


Vemos que todos los valores de las columnas son objetc por que pasamos todos los valores menos fechas a float

In [17]:
df_1[["temp_max_cantabria", "temp_min_cantabria", "precipitaciones_cantabria","temp_max_galicia","temp_min_galicia","precipitaciones_galicia","temp_max_navarra","temp_min_navarra","precipitaciones_navarra","temp_max_andalucia",	"temp_min_andalucia","precipitaciones_andalucia"]] = df_1[["temp_max_cantabria", "temp_min_cantabria", "precipitaciones_cantabria","temp_max_galicia","temp_min_galicia","precipitaciones_galicia","temp_max_navarra","temp_min_navarra","precipitaciones_navarra","temp_max_andalucia","temp_min_andalucia","precipitaciones_andalucia"]].astype(float)

In [18]:
df_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 13 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   fechas                     12 non-null     object 
 1   temp_max_cantabria         12 non-null     float64
 2   temp_min_cantabria         12 non-null     float64
 3   precipitaciones_cantabria  12 non-null     float64
 4   temp_max_galicia           12 non-null     float64
 5   temp_min_galicia           12 non-null     float64
 6   precipitaciones_galicia    12 non-null     float64
 7   temp_max_navarra           12 non-null     float64
 8   temp_min_navarra           12 non-null     float64
 9   precipitaciones_navarra    12 non-null     float64
 10  temp_max_andalucia         12 non-null     float64
 11  temp_min_andalucia         12 non-null     float64
 12  precipitaciones_andalucia  12 non-null     float64
dtypes: float64(12), object(1)
memory usage: 1.3+ KB


### 5.3 Creamos los gráficos necesarios

Importamos las librerías necesarias para mostrar los valores en gráficos, utilizaremos para ello Plotly

In [19]:
import ipywidgets as widgets
import matplotlib.pyplot as plt
import plotly.graph_objects as go

#### 5.3.1 Gráfico: Media de temperaturas máximas

Para realizar la primera comparativa de forma visual, compararemos la media de las temperaturas máximas de cada Comunidad. 

In [34]:
fig = go.Figure()

fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["temp_max_cantabria"], name="Cantabria", marker_color="purple"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["temp_max_galicia"], name="Galicia", marker_color="green"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["temp_max_navarra"], name="Navarra", marker_color="yellow"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["temp_max_andalucia"], name="Andalucia", marker_color="pink"))

# Configura el layout
fig.update_layout(title="Media de temperaturas máximas por Comunidad",
                  xaxis_title="Meses",
                  yaxis_title="Temperatura en Cº",
                  barmode='group')

fig.show()

- Gráfico 1. Media de temperaturas máximas por Comunidad

#### 5.3.2 Gráfico: Media de temperaturas mínimas

Ahora pasaremos a graficar la media de las temperaturas mínimas.

In [35]:
fig = go.Figure()

fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["temp_min_cantabria"], name="Cantabria", marker_color="purple"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["temp_min_galicia"], name="Galicia", marker_color="green"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["temp_min_navarra"], name="Navarra", marker_color="yellow"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["temp_min_andalucia"], name="Andalucia",  marker_color="pink"))

# Configura el layout
fig.update_layout(title="Media de temperaturas mínimas por Comunidad",
                  xaxis_title="Meses",
                  yaxis_title="Temperatura en Cº",
                  barmode='group')

fig.show()

- Gráfico 2. Media de temperaturas mínimas por Comunidad

#### 5.3.3 Gráfico: Media de precipitaciones

In [36]:
fig = go.Figure()

fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["precipitaciones_cantabria"], name="Cantabria",  marker_color="purple"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["precipitaciones_galicia"], name="Galicia",  marker_color="green"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["precipitaciones_navarra"], name="Navarra",  marker_color="yellow"))
fig.add_trace(go.Bar(x=df_1["fechas"], y=df_1["precipitaciones_andalucia"], name="Andalucia",  marker_color="pink"))

# Configura el layout
fig.update_layout(title="Media de precipitaciones por Comunidad",
                  xaxis_title="Meses",
                  yaxis_title="Temperatura en Cº",
                  barmode='group')

fig.show()

- Gráfico 3. Media de precipitaciones por Comunidad

In [23]:
df_1.describe()

temp_max_cantabria  temp_min_cantabria  precipitaciones_cantabria  \
count           12.000000           12.000000                  12.000000   
mean            18.616667           13.383333                  28.700000   
std              4.182177            4.189127                  25.927767   
min             12.700000            7.600000                   8.200000   
25%             15.475000           10.225000                  13.125000   
50%             17.800000           12.800000                  17.400000   
75%             22.325000           17.650000                  33.350000   
max             24.000000           18.500000                  94.600000   

       temp_max_galicia  temp_min_galicia  precipitaciones_galicia  \
count         12.000000         12.000000                12.000000   
mean          19.550000         13.358333                19.083333   
std            4.010554          3.750990                15.890868   
min           13.700000          7.100000                 3.800000   
25%           16.425000         10.700000                 9.450000   
50%           19.550000         13.200000                14.500000   
75%           23.300000         16.975000                24.650000   
max           24.700000         18.300000                56.800000   

       temp_max_navarra  temp_min_navarra  precipitaciones_navarra  \
count         12.000000         12.000000                12.000000   
mean          20.425000          8.033333                23.075000   
std            7.756302          5.989739                16.030716   
min            8.400000         -1.500000                 4.100000   
25%           14.425000          3.375000                11.625000   
50%           20.650000          7.700000                21.550000   
75%           27.175000         14.000000                29.925000   
max           31.800000         15.800000                62.600000   

       temp_max_andalucia  temp_min_andalucia  precipitaciones_andalucia  
count           12.000000           12.000000                  12.000000  
mean            25.916667           13.466667                  13.633333  
std              6.411614            5.449993                  19.130144  
min             17.200000            6.300000                   0.000000  
25%             21.250000            8.800000                   3.400000  
50%             27.050000           13.700000                   5.700000  
75%             29.500000           17.700000                  15.600000  
max             36.600000           21.000000                  67.600000

In [26]:
df_1.loc[:, "precipitaciones_cantabria"].sum()

np.float64(344.4)

In [27]:
df_1.loc[:, "precipitaciones_navarra"].sum()

np.float64(276.9)

In [28]:
df_1.loc[:, "precipitaciones_galicia"].sum()

np.float64(229.00000000000003)

In [29]:
df_1.loc[:, "precipitaciones_andalucia"].sum()

np.float64(163.59999999999997)

### 6. CONCLUSIÓN

Media de temperatura máxima en Cantabria (2023): 18,61°

Media de temperatura mínima en Cantabria (2023): 13,38°

Total de las precipitaciones en milímetros (2023): 344,4 mm

Media de temperatura máxima en Galicia (2023):  19,55°

Media de temperatura mínima en Galicia (2023): 13,36°

Total de las precipitaciones en milímetros (2023): 229 mm

Media de temperatura máxima en Navarra (2023):  20,45°

Media de temperatura mínima en Navarra (2023): 8,77°

Total de las precipitaciones en milímetros (2023): 276,9 mm

Media de temperatura máxima en Andalucía (2023): 27,83°

Media de temperatura máxima en Andalucía (2023): 13,87°

Total de las precipitaciones en milímetros(2023): 163,59 mm

Hemos observado que Andalucía presenta las temperaturas más altas, lo que hace que esta región sea un área clave para centrarnos en nuestro estudio. En primer lugar, la diferencia de precio por metro cuadrado entre las zonas urbanas y el suelo urbano es casi la mitad, lo que representa una ventaja económica. Además, la climatología de la región, con bajas precipitaciones y temperaturas estables, es ideal para el tipo de producto que nuestro cliente comercializa.

No obstante, también estamos enfocando nuestra atención en las Comunidades del Norte, donde, si bien las temperaturas y precipitaciones son similares entre sí, lo que hará que elijamos una u otra en este caso, ya no va a ser la temperatura, sino la diferencia del precio por metro cuadrado. En este sentido, seleccionamos Cantabria, ya que presenta la mayor variación en el precio por metro cuadrado entre la ciudad y los municipios cercanos.

Por lo que, para el siguiente análisis (3_disponibilidad_suelo_urbanizable), seleccionamos las Comunidades de Cantabria y Andalucía